### ISSUE TITLE
We should use templates for issues!

### ISSUE THREAD

**Issue by obw** [olivierkes/manuskript#1217]
In the Issue #1216 we had a lot of Communication, for gathering the needed parameters too get to the point, where we can try to fix the bug!

I also use TexStudio for working on my stories and have created an issue there, they have Templates to Help the user, to give the needed information, to track down the Error!

You can take a look here, how it's looking: https://github.com/texstudio-org/texstudio/issues/new/choose

I don't know, how to make these, also I don't have the rights, but I think, that's something, which could help us, to give the User a better experience!

Regards!

**Comment by TheJackiMonster** [comment]
I think templates on Github always come with advantages and disadvantages. Because while they can ease gathering proper information, they rarely contain everything you need. Then people increase complexity of these templates trying to solve this issue but all it does is preventing users from opening issues.

I had multiple cases of me wanting to open an issue in other projects. But their templates were so convoluted that I ended not doing anything instead. Most of the times these templates ask for information which has nothing to do with any of your issue.

I personally think you should not need templates since issues won't be solved if people don't provide the required information to do so. If an issue lacks communication of such information, we can still close it in worst case because of inactivity. But since bugs can potentially be reproduced by multiple people even other users could provide the needed information the original author of an issue did not provide. So I usually leave issues open if anything is missing.

**Comment by obw** [comment]
@TheJackiMonster it should be easy to generate a command line Parameter, like --getissueinfo

Output something like that:

OS: Linux 5.15.0-83-generic #92-Ubuntu SMP Mon Aug 14 09:30:42 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

This Information we have in the About Dialog:
Manuskript Version: 0.15.0
Python: 3.10.12
PyQt: 5.15.6
Qt: 5.15.3

And then we only need a Log-File when it's a crash....

The Template could look like something like that:

# I have a Crash Template

Please run `manuskript --getissueinfo` from the Console or copy the Info about your System out `Help` -> `About`

Then run `manuskript --logfile mylog.log` and add mylog.log to the issue, so that we can better find your problem...

-----

Something like that!

**Comment by TheJackiMonster** [comment]
@obw Since I tend to recommend everyone commenting on the website about issues to write about them here on Github instead for proper tracking, I don't think that makes it very easy to use for many people.

I don't think most people would use a command line interface for generating an issue on Github. That's just not intuitive. So we would likely just end up enforcing a template, most people don't know how to generate. Then we close all opened issues, point to an explanation on the website or in the wiki but it's still not clear whether everyone will be open to get through this process. I think in the end we would just cause a lot of friction, likely getting less issues and therefore less feedback.

Don't get me wrong. There are benefits in such templates but I don't think it makes sense for every project. Especially within open-source project like Manuskript I don't expect users to keep up with standards when posting about their problems. The same way no user should expect their issue to be solved. Eventually someone might take a look and solve it but that's pretty much the deal since there's no payment involved on any end.

**Comment by obw** [comment]
@TheJackiMonster I know what are you thinking... and also you are right! But my thinking behind it, is to make it easier for all parties to get to a good result!

Something, which has been come to my mind, after this discussion so far, we should add a Menu Point:

`Help` -> `Information for Issue to clipboard`

Because a lot of Users don't know / register that the Text in `Help` -> `About` can be marked, and so copied...

Also, we could get some Information over the OS, the User can remove this from the Issue Text, when he didn't want to share it!

This is an Open-Source Project, but also we need help to make it better! I have tried to come over my allergic reaction too python as programming language, but sorry no chance! Python and I will never be friend and I don't know why!

I will restart my effort, for making a complete Documentation, when the GTK-Port is complete! There will be so much changes, that this would be Work for nothing! See #726!

Perhaps, I will add what I have the next days to GitHub.

Regards!

**Comment by peter88213** [comment]
Why not activate the "Discussions" forum right away? 

Many "issues" here could be moved there, and feature requests could be discussed there first before turning them into an issue (which is easy). 

The forum offers the possibility of setting up categories in which users who are not developers themselves can find their way around, and you would have the chance to establish something like a user community. 
The open issues would be fewer and you could introduce stricter rules. 

Merry Christmas, by the way!

**Comment by TheShadowOfHassen** [comment]
I agree with @peter88213  a ton of issues we get are either feature requests or some bug report for an old version that was fixed years ago.  There are at least 20 open issues that are like that. I'm not a big fan of issue templates but if the feature requests were moved to the forums and there was a reminder when you opened an issue to make sure your software was updated we'd have a third of our open issues

**Comment by TheJackiMonster** [comment]
I'm not convinced that opening a forum decreases the amount of issues. Adding a forum would move discussions from the issues into the forum. So information would be split between multiple areas which would actually increase complexity for developers to gather all important aspects from users for an implementation.

I think reducing the amount of issues on Github isn't even a proper goal to begin with. I don't mind open issues as long as they still contain valid information for future contributions. The amount of issues does not matter really. Everyone who wants to contribute can simply pick any issue, look into it, implement or comment on it to solve it.

The most reason for old open issues is that they have close to zero traction. But a forum would not change that. Forums tend to hide old posts in a very similar way. It would just look cleaner in the beginning because old issues wouldn't automatically move over. But I don't see how this changes actual progress.

Users can already discuss in the issues. I also tag all issues so that developers and users can filter to contribute to their specific needs. For example developers can easily search for bugs or feature requests and users can search for questions and discussions to help each other. Your argument now is that a forum would split those topics properly. But in reality we can only assume how such a forum would be used. So as developer I would now look through issues and forum to find information for a certain feature request, making my life more difficult.
